### PR TITLE
🙈 reCAPTCHA owners

### PR DESCRIPTION
**Pull request by andycochran**
This PR… 
- Adds the `ember-g-recaptcha` addon 
  - The addon requires node 10, so support for node 8 is dropped
  - Since Netlify locks the node version you're already using, a new `.node-version` file explicitly changes it
- Adds a  `ENV.gReCaptcha.siteKey` to the config
- Adds reCAPTCHA to owner field on tax lots with a new `<RecaptchaBlock>` component,  which: 
    - Hides its block content
    - Shows a button by default (with the passed `@buttonText`)
    - Shows a spinner and tries to prove you're not a robot on click
    - Then reveals the content after reCAPTCHA validates
- Adds test coverage for `recaptcha-block` 

NOTES:
- The `siteKey` in the `test` environment commented out (there in case we need it in the future) 
- The `<RecaptchaBlock>` component is built to work w/o `ember-g-recaptcha`, and explicitly excludes it in the `test` environment. Without doing so, all the tests that load a lot break because they don't wait for recaptcha to do its thing. Even using the test key (which skips ajax/validation) race conditions abound. We'd have to put `waitFor` in every lot test to prevent `ember-g-recaptcha` getting torn down before it runs. 
- Validation is encapsulated in the widget. The `reCaptchaResponse` token is proof that the user completed the bot check. However, it does not prevent HTTP client request forgery. That's not a problem for this app. But if we ever need to use reCAPTCHA with extra-strong security, we can validate the token with a `POST` to `https://www.google.com/recaptcha/api/siteverify?secret=[secretkey]&response=[token]` and check the response: `{ "success": [true|false], ... }`.

### CODE REVIEW

**Comment by andycochran** [comment]
I think I can ditch the `reCaptchaRendered`. I don't think it's doing anything now.

File: app/components/recaptcha-block.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  @action
+  onCaptchaResolved(reCaptchaResponse) {
+    this.set('reCaptchaResponse', reCaptchaResponse);
+    this.set('reCaptchaWaiting', false);
+  }
+
+  @action
+  resolveCaptcha() {
+    this.set('reCaptchaWaiting', true);
+    const executeRecaptcha = this.isTestEnvironment ? () => {} : window.grecaptcha.execute();
```
[the remark this answers is not in the document]
Comment (allthesignals) [reply app/components/recaptcha-block.js:17]: on second thought, i'd just add more code to make this easier to grock:

```javascript
if (this.isTestEnvironment) {
  // do nothing!
} else {
  window.grecaptcha.execute();
}
```

File: tests/unit/components/recaptcha-block-test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+module('Unit | Component | recaptcha-block', function(hooks) {
+  setupTest(hooks);
+
+  // Replace this with your real tests.
+  test('its actions update its args', function(assert) {
+    const component = this.owner.factoryFor('component:recaptcha-block').create();
```
Comment (allthesignals) [inline tests/unit/components/recaptcha-block-test.js:9]: Nice! You mock a class we own, and that's fine.

I will point out that mocking in unit tests, to this extent, makes the mocked class resistent to refactoring (in my experience).

File: app/components/recaptcha-block.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (andycochran) [reply app/components/recaptcha-block.js:17]: TBH, I had it in a multiline if as you recommend, but changed it to a ternary just so codecov would hit all lines. 😬 

(I'm gonna investigate the addon's test support a bit more and see if I can avoid the ENV sniffing.)

[the remark this answers is not in the document]
Comment (allthesignals) [reply app/components/recaptcha-block.js:17]: Hmm, codecov should still ding you on "partial line coverage", though...

File: config/environment.js [hunk 3]
```diff
@@ -297,6 +302,8 @@ module.exports = function(environment) {
     // Testem prefers this...
     ENV.locationType = 'none';
 
+    // ENV.gReCaptcha.siteKey = '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI';
```
[the remark this answers is not in the document]
Comment (andycochran) [reply config/environment.js:305]: I think I'll go ahead and delete it. It's not _our_ key. It's a generic key for automated testing, which we can easily grab again if we need it. https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-tests-with-recaptcha.-what-should-i-do

**Pull request merged**